operations/curseforge/client.go

```diff
@@ -39,29 +40,66 @@ func getLatestFiles(projectId uint) ([]File, error) {
 	if err != nil {
 		return nil, err
 	}
-	return addon.Data.LatestFiles, err
+	return addon, nil
 }
 
-func getFileById(projectId, fileId uint) (File, error) {
+func getAddonFileData(projectId uint, fileId uint) (FileRespnse, error)
+{
 	u := fmt.Sprintf("https://api.curseforge.com/v1/mods/%d/files/%d", projectId, fileId)
 
 	response, err := callCurseForge(u)
 	if err != nil {
-		return File{}, err
+		return nil, err
```

Comment:
File response is a struct, so it can't be nullable (unless it is a pointer, which this isn't)